gateway/src/auth.rs

```diff
@@ -32,18 +32,15 @@ where
 {
     type Rejection = Error;
 
-    async fn from_request_parts(parts: &mut Parts, state: &S) -> Result<Self, Self::Rejection> {
+    async fn from_request_parts(parts: &mut Parts, _state: &S) -> Result<Self, Self::Rejection> {
         let claim = parts.extensions.get::<Claim>().ok_or(ErrorKind::Internal)?;
         let user_id = claim.sub.clone();
```

Comment:
Does it make sense to still have a `User` struct? We can just implement `FromRequestParts` for the claim right?

Reply:
*only traits defined in the current crate can be implemented for a type parameter*
Claim can technically be moved to backends (it seems), but I'd skip it for now.